Flutter/casi/lib/casi_user.dart

```diff
@@ -1,134 +1,205 @@
+    // );                                                                                               //by CASI on posting the token.
+    // final jsonData = json.decode(response.body);                                                     //test
+    // return CasiUser.fromJson(jsonData['user']);                                                      //test
+    
+    final response = await http.post(                                                             //actual                
+        Uri.parse(this._serverUrl + '/auth/refresh-token'),                                       //actual
+        body: {'rememberme': toSendToken});                                                       //actual
+    final jsonData = jsonDecode(response.body);                                                   //actual
+    if (response.statusCode == 200) {                                                             //actual
+      if (onRefreshSuccess != null) {                                                             //actual
+        onRefreshSuccess(                                                                         //actual
+            response.headers['set-cookie']!.split(';')[0].split('=')[1]);                         //actual
+      }                                                                                           //actual
+      return CasiUser.fromJson(jsonData['user']);                                                 //actual
+    } else {                                                                                      //actual
+      throw Exception(jsonData['msg']);                                                           //actual
+    }                                                                                             //actual
+  }
+}
+
+class Webview extends ChromeSafariBrowser {
+  String getTokenURL;
+  Function fetchUserDetails;
+  Function onSuccess;
+  Function onError;
+  String secret;
+  Function? loading;
+  Webview(
+      {
+      required this.getTokenURL,
+      required this.fetchUserDetails,
+      required this.onSuccess,
+      required this.onError,
+      required this.secret,
+      required this.loading,
       }
-      return CasiUser.fromJson(jsonData['user']);
-    } else {
-      throw Exception(jsonData['msg']);
+  );
+
+  @override
+  void onClosed() async {
+    try {
+      loading!(true);
+      var tokenResponse = await http.post(Uri.parse(getTokenURL),
+          body: json.encode({
+            'key1': 'val1',                                      //POST request body to get oauth token
+            'key2': 'val2',
+          }));
+
+      String token = json.decode(
+          tokenResponse.body)['name'];                           //Enter json key of token in response
+      print(token);
+      print("ok");
+      CasiUser user = await fetchUserDetails(token);
+      loading!(false);
+      print("x");
```

Comment:
If you are done testing all of this, can you please remove all such unnecessary print statements before we merge this?

Reply:
Yeah I was supposed to. Will remove them.